File: kt03/src/main/java/com/btime/kt03/instance/java/LazyThreadSafeSynchronized.java
package com.btime.kt03.instance.java;

public class LazyThreadSafeSynchronized {
    private static LazyThreadSafeSynchronized INSTANCE;

    private LazyThreadSafeSynchronized(){}

    public static synchronized LazyThreadSafeSynchronized getInstance(){
        if(INSTANCE == null){
            INSTANCE = new LazyThreadSafeSynchronized();
        }
        return INSTANCE;
    }
}

File: kt03/src/main/java/com/btime/kt03/instance/kotlin/LazyThreadSafeSynchronized.kt
package com.btime.kt03.instance.kotlin

class LazyThreadSafeSynchronized private constructor(){
    companion object{
        private var instance:LazyThreadSafeSynchronized ?= null

        @Synchronized
        fun get():LazyThreadSafeSynchronized{
            if(instance == null) instance = LazyThreadSafeSynchronized()
            return instance!!
        }
    }
}